### PR TITLE
File Header Check/Diff Reporter + Companion Git Action

### PR DESCRIPTION
**Pull request by dachalco**
Description
-----------
+ New script that checks file headers for input file(s). Shows familiar diff. Can accept list of files in `*.json` format too.
+ Git action that is triggered by pull requests. It can be added as a mandatory check so we can enforce file headers (version, copyright, licensing)

Test Steps
-----------
Run the script locally. See `check-header.py -h`.
Can also trigger git action by adding new PR.

Related Issue
-----------
n/a

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by dachalco** [comment]
FYI, it's failing it's own check as it tries to run `FreeRTOS/FreeRTOS/.github/scripts/check-header.py` which, of course, does not exist yet.

It's interesting that git is workflows as defined in the PR git ref, versus what is actually in `master`.

File: .github/scripts/check-header.py [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+        This is particularly useful when ingesting output from other programs, like git actions
+        '''
+        assert os.path.exists(path_json), 'No such file: ' + path_json
+
+        # Get list of files to check from JSON file
+        with open(path_json) as file_json:
+            file_checklist = load(file_json)
+            assert isinstance(file_checklist, list), 'Expected list for singular JSON List entry'
+
+        # Accrue how how files fail the check
```
Comment (cobusve) [inline .github/scripts/check-header.py:30]: how how

File: .github/scripts/check-header.py [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+            else:
+                print('File Delta: %s' % path)
+                print(*unified_diff(lines[:len(self.header)], self.header))
+                return False
+
+
+def configArgParser():
+    parser = ArgumentParser(description='FreeRTOS file header checker. We expect a consistent header across all '
+                                        'first party files. The header includes current version number, copyright, '
+                                        'and FreeRTOS license.')
```
Comment (cobusve) [inline .github/scripts/check-header.py:56]: Are we also checking the 2 links?

**Review by cobusve — APPROVED** [review]

File: .github/scripts/check-header.py (hunk 2 above)
Comment (dachalco) [reply .github/scripts/check-header.py:56]: Yes, for now the expected `freertos_header` and `kernel_header` are set in main and include version number, copyright, license, and the two links. If anything diffs here, it reports a diff similar looking to gitdiff. Additionally any diffs in any of the checked files will fail the git action

File: .github/scripts/check-header.py (hunk 1 above)
Comment (dachalco) [reply .github/scripts/check-header.py:30]: @dachalco does not compute

Comment (dachalco) [reply .github/scripts/check-header.py:30]: do you mean how is loading? If so, at top it `from json import load`.
If the JSON formatting of the loaded file is in question, perhaps I should add more detail to the assert message.
Currently the JSON intakes exclusively exists for cooperation with:
https://github.com/lots0logs/gh-action-get-changed-files

**Pull request merged**